src/core/Akka.Remote.TestKit/MultiNodeSpec.cs

```diff
@@ -506,6 +506,16 @@ public void RunOn(Action thunk, params RoleName[] nodes)
             if (nodes.Length == 0) throw new ArgumentException("No node given to run on.");
             if (IsNode(nodes)) thunk();
         }
+        
+        /// <summary>
+        /// Execute the given block of code only on the given nodes (names according
+        /// to the `roleMap`).
+        /// </summary>
+        public async Task RunOnAsync(Func<Task> thunkAsync, params RoleName[] nodes)
```

Comment:
Well - this particular overload I added because `thunk` node action was containing some of `Akka.TestKit` methods, which I was using in new `async` variation. So to use `AwaitAssertAsync` and others we need to have this `RunOnAsync` overload.
This is the most popular blocking methods. Another blocking methods are about event filtering (i.e. `ExpectMsg`), and they turned out to be using `BlockingCollection` internally - so no way to easily convert them to `async` methods without changing the event capturing implementation.
We may add `async` overloads to all (I mean to less popular as well) test methods that use blocking API anyway, for example `EnterBarrier` and other. Well, `EnterBarrier` may block bunch of threads during execution actually (several nodes, several parallel tests, every node is waiting for others on the barrier...) Should I create issue for this?

Reply:
Yes, create an issue with making an async overload for EnterBarrier. If we decide to implement it we will want to convert all of our MNTR specs to return `Task` as well.

Reply:
Done: https://github.com/akkadotnet/akka.net/issues/4146